aten/src/ATen/native/cuda/Dropout.cu

```diff
@@ -80,20 +78,31 @@ fused_dropout_kernel_vec(at::cuda::detail::TensorInfo<const scalar_t, IndexType>
     //curand_uniform_double was pure evil anyway, not doing what it promises, and there's nothing for halfs, so generate float for everything
     // Note: need a new set of random values per 4 elements -- we'll handle VEC elements in this thread, so need ceil(VEC / 4)
     // sets of rand.
-    if ((VEC == 4) || (gridxvec_loop_state == 0)) {
-      rand = curand_uniform4(&state);
+    if ((VEC >= 4) || (gridxvec_loop_state == 0)) {
+      #pragma unroll
```

Comment:
Will this affect non-ROCM backends?

Reply:
For non-ROCM backends, VEC will be 2 or 4. So it won't change the behavior.
I have added an explicit assert to ensure that.